From da4b198dcbcfb71ed5cbc2d613943d7560a53931 Mon Sep 17 00:00:00 2001
From: Sean Mackrory <sean@cloudera.com>
Date: Tue, 17 Jan 2017 13:33:45 -0700
Subject: [PATCH 2253/2748] CLOUDERA-BUILD. Changes to make s3a CLI command
 built-in in CDH

The way to do this differs between Hadoop 2 and 3, so it is orthogonal
to any other patch.

Change-Id: Ibcdbb6eb0895819d19a67651013b041f99fa4648
---
 .../hadoop-common/src/main/bin/hadoop              |    4 ++++
 1 file changed, 4 insertions(+)

diff --git a/hadoop-common-project/hadoop-common/src/main/bin/hadoop b/hadoop-common-project/hadoop-common/src/main/bin/hadoop
index eeab027..7a96663 100755
--- a/hadoop-common-project/hadoop-common/src/main/bin/hadoop
+++ b/hadoop-common-project/hadoop-common/src/main/bin/hadoop
@@ -38,6 +38,7 @@ function print_usage(){
   echo "  credential           interact with credential providers"
   echo "                       Hadoop jar and the required libraries"
   echo "  daemonlog            get/set the log level for each daemon"
+  echo "  s3a                  manage data on S3"
   echo "  trace                view and modify Hadoop tracing settings"
   echo " or"
   echo "  CLASSNAME            run the class named CLASSNAME"
@@ -115,6 +116,9 @@ case $COMMAND in
       CLASSPATH=${CLASSPATH}:${TOOL_PATH}
     elif [ "$COMMAND" = "credential" ] ; then
       CLASS=org.apache.hadoop.security.alias.CredentialShell
+    elif [ "$COMMAND" = "s3a" ] ; then
+      CLASS=org.apache.hadoop.fs.s3a.s3guard.S3GuardTool
+      CLASSPATH=${CLASSPATH}:${TOOL_PATH}
     elif [ "$COMMAND" = "trace" ] ; then
       CLASS=org.apache.hadoop.tracing.TraceAdmin
     elif [ "$COMMAND" = "classpath" ] ; then
-- 
1.7.9.5

